### ISSUE TITLE
An issue about the default initialize methods.

### ISSUE THREAD

**Issue by LiChenda** [espnet/espnet#4555]
Hello, I'm training some custom models with ESPNet2. And I find that the default initialize method checking the bias parameters of NN by the  `p.dim()` function. This check is not precise (some parameters my also make `p.dim() == 1` be `True`  ) and set them to zero may lead to abnormal training. (I found one of my module always output zeros and the training may dead with ReLU. After removing the following lines, the training becomes OK. )

https://github.com/espnet/espnet/blob/96bd74641ceb463096067223d0734f70bddd8def/espnet2/torch_utils/initialize.py#L77-L80

**Comment by sw005320** [comment]
@LiChenda, thanks for the report.
Can you tell me more examples of when some parameters also make `p.dim() == 1` be `True`?
How about making your treatment an option?

**Comment by LiChenda** [comment]
Some parameters like the `weight` $\gamma$  in `BatchNorm`,  and some custom parameters defined with `torch.nn.Parameter()` may make `p.dim() == 1` be `True`.
One possible solution might be updating these lines https://github.com/espnet/espnet/blob/96bd74641ceb463096067223d0734f70bddd8def/espnet2/torch_utils/initialize.py#L77-L80
 with:
```python

for name, p in model.named_parameters(): 
    if 'bias' in name:
        p.data.zero_() 

```

**Comment by LiChenda** [comment]
BTW, this function is called by amlost all the tasks in ESPnet2, I'm not sure if these updating keeps the previously trained model reproducible.

**Comment by sw005320** [comment]
> ```python
> for name, p in model.named_parameters(): 
>     if 'bias' in name:
>         p.data.zero_() 
> ```

This sounds good to me.
Yes, we can make this an option and make default false in some timing.
Meanwhile, we can conduct some tests and make default true.
Can you make a PR?

**Comment by LiChenda** [comment]
Sure!

**Comment by popcornell** [comment]
```python
for name, p in model.named_parameters(): 
    if 'bias' in name:
        p.data.zero_() 
```

maybe make it case insensitive too just to be sure. 
This is difficult to implement in a scalable way. 

Maybe raising a warning also could be useful so user knows that the bias is set to zero.  But warning should be likely raised once (e.g. for each layer of that class) otherwise will be too verbose maybe.

**Comment by b-flo** [comment]
> BTW, this function is called by amlost all the tasks in ESPnet2, I'm not sure if these updating keeps the previously trained model reproducible.

It should be noted that `args.init` is set to `None` by default and most configs don't use initialization.
We currently have **1026** configs (!!). 172 configs use either `chainer` or `xavier_uniform` initialization, from which **110** are for the ASR task and **59** for the ENH task.
From what I see, it doesn't seem to impact config we usually use, at least for ASR.

**Comment by LiChenda** [comment]
> > BTW, this function is called by amlost all the tasks in ESPnet2, I'm not sure if these updating keeps the previously trained model reproducible.
> 
> It should be noted that `args.init` is set to `None` by default for all tasks and most configs don't use initialization. We currently have **1026** configs (!!). **172** configs use either `chainer` or `xavier_uniform` initialization, from which **110** are for the ASR task and **59** for the ENH task. From what I see, it doesn't seem to impact config we usually use, at least for ASR.

Thanks for pointing it out. I just made a PR for this issue, see #4574 .

**Comment by LiChenda** [comment]
The PR is merged, so I close this issue.

**Comment by kan-bayashi** [comment]
I found this issue has an impact on the most of TTS models since TTS modules is initialized using this method as a default.
https://github.com/espnet/espnet/blob/14fcb2d42b2609f766ffaa7a79e9c921cd8398d9/espnet2/tts/fastspeech2/fastspeech2.py#L470-L475
https://github.com/espnet/espnet/blob/14fcb2d42b2609f766ffaa7a79e9c921cd8398d9/espnet2/tts/fastspeech2/fastspeech2.py#L828-L829

The model including `BatchNorm2d` or `BatchNorm1d` modules should be improved with this changes.
(I'm not sure what is happened before fixing this issue...)
